### PR TITLE
roachtest: skip backup and restore tests that flake due to network error

### PR DESCRIPTION
**Pull request by msbutler**
As of 24.1, backup and restore are more resilient to network flakes due to
 #116957 and #119804. These patches were not backported to 23.2. To reduce test
flake noise in 23.2 and earlier branches, this patch skips a roachtest where backup/restore paused or failed due to a network flake.

Epic: none

Release note: none

### CODE REVIEW

File: pkg/cmd/roachtest/tests/restore.go [hunk 1]
```diff
@@ -478,6 +482,35 @@ func registerRestore(r registry.Registry) {
 	}
 }
 
+// skipIfNetworkFlake skips the test if the job error contains a network flake.
+// This function should only get called after a job terminates unexpectedly.
+func skipIfNetworkFlake(ctx context.Context, t test.Test, c cluster.Cluster, jobId int) error {
+	db, err := c.ConnE(ctx, t.L(), c.Node(1)[0])
```
Comment (stevendanna) [inline pkg/cmd/roachtest/tests/restore.go:488]: Should we `defer db.Close()` here?

File: pkg/cmd/roachtest/tests/restore.go [hunk 2]
```diff
@@ -478,6 +482,35 @@ func registerRestore(r registry.Registry) {
...
+
+	networkFlakes := []string{
+		"http2: client connection force closed via ClientConn.Close",
+		"HTTP response code 416",
+		"tls: bad record MAC",
+	}
+
+	for _, flake := range networkFlakes {
+		if strings.Contains(jobError, flake) {
+			t.Skipf("skipping test because of network flake", jobError)
```
Comment (stevendanna) [inline pkg/cmd/roachtest/tests/restore.go:508]: Did you mean 
```suggestion
			t.Skipf("skipping test because of network flake: %s", jobError)
```

Comment (msbutler) [reply pkg/cmd/roachtest/tests/restore.go:508]: nice catch. done.

File: pkg/cmd/roachtest/tests/restore.go (hunk 1 above)
Comment (msbutler) [reply pkg/cmd/roachtest/tests/restore.go:488]: the go compiler does not appreciate your suggestion:
```
compilepkg: nogo: errors found by nogo during build-time code analysis:
pkg/cmd/roachtest/tests/restore.go:489:2: should check returned error before deferring db.Close() (SA5001)
```

**Review by stevendanna — APPROVED** [review]

**Pull request merged**